### PR TITLE
Expand explanation articles

### PR DESCRIPTION
**Pull request by kamilzyla**
### Changes
Addresses #192, but does not close it. Expand articles:
1. Explanation: Limitations and known issues.
2. Explanation: Node.js.
3. Explanation: Project structure.

They should now be understandable by our colleagues / suitable for internal use in Appsilon, but might require further editing. I will create issues for editing once this PR is merged. 

### How to test
Run `pkgdown::build_site()` and review the expanded articles.

### CODE REVIEW

File: vignettes/explanation-limitations-and-known-issues.Rmd [hunk 1]
```diff
@@ -7,11 +7,39 @@ vignette: >
   %\VignetteEncoding{UTF-8}
 ---
 
-A Rhino app must live in git root for GitHub Actions CI to work.
+# GitHub Actions CI
```
Comment (marekrogala) [inline vignettes/explanation-limitations-and-known-issues.Rmd:10]: I don't like this explanation. I don't think it actually explains a specific topic.... And unnecessarily sounds bad.

Instead, i suggest:
GitHub Actions CI --> How to: Rhino app outside repository root
Box --> Explanation: Using box modules

File: vignettes/explanation-node-js.Rmd [hunk 2]
```diff
@@ -7,11 +7,37 @@ vignette: >
   %\VignetteEncoding{UTF-8}
 ---
```
Comment (marekrogala) [inline vignettes/explanation-node-js.Rmd:8]: Explanation: Node.js -->. Explanation: How Rhino builds CSS and JS using Node

File: vignettes/explanation-project-structure.Rmd [hunk 3]
```diff
@@ -63,18 +63,29 @@ server <- function(id) {
...
-so by keeping the `app.R` minimal and loading the whole app with box
-we can be confident that it is properly used throughout the app.
-
-Additionally, `rhino:app()` gives us full control over app startup. It does:
-1. Purge box cache (reload app without restarting R)
-2. Configure logger (level, file)
-3. Configure static files
-4. Load main module / legacy entrypoint
-5. Add head tags (favicon, CSS & JS)
+# Minimal `app.R`
```
Comment (marekrogala) [inline vignettes/explanation-project-structure.Rmd:66]: I suggest a title like "Explanation: application structure in Rhino"

**Review by marekrogala — APPROVED** [review]
LGTM, a few comments 💯

File: vignettes/explanation-limitations-and-known-issues.Rmd (hunk 1 above)
Comment (kamilzyla) [reply vignettes/explanation-limitations-and-known-issues.Rmd]: Good point, I split the article.

File: vignettes/explanation-node-js.Rmd (hunk 2 above)
Comment (kamilzyla) [reply vignettes/explanation-node-js.Rmd]: Expanded the title.

File: vignettes/explanation-project-structure.Rmd (hunk 3 above)
Comment (kamilzyla) [reply vignettes/explanation-project-structure.Rmd]: Renamed to "Explanation: Application structure". I decided to leave "Rhino" out - I think its obvious.

**Pull request merged**